lib/jison.js

```diff
@@ -1280,7 +1280,12 @@ function commonjsMain (args) {
         console.log('Usage: '+args[0]+' FILE');
         process.exit(1);
     }
-    var source = require('fs').readFileSync(require('path').normalize(args[1]), "utf8");
+    var source = null;
+    var fs = require('fs');
+    var path = require('path');
+    if (typeof fs !== 'undefined' && fs !== null && typeof path !== 'undefined' && path !== null) {
+        source = fs.readFileSync(path.normalize(args[1]), "utf8");
+    }
     return exports.parser.parse(source);
```

Comment:
This is getting included inside the browser-based CoffeeScript compiler. The use case for these checks is when someone is using the compiler in a project that has require.js, which basically simulates a CommonJS environment. See https://github.com/jashkenas/coffeescript/issues/4391. I’m trying to make things not crash in such an environment.

Perhaps the `return exports.parser.parse(source);` could be inside the `if`, and this function returns nothing when `readFileSync` is unavailable; I didn’t try that. I’m under the impression though that downstream tools need whatever this function is returning, even if it’s a parsing of an empty string.